tests/components/pooldose/test_config_flow.py

```diff
@@ -0,0 +1,433 @@
+"""Test the Seko Pooldose config flow."""
+
+from unittest.mock import AsyncMock, MagicMock, patch
+
+from pooldose.request_status import RequestStatus
+import pytest
+
+from homeassistant.components.pooldose.const import DOMAIN
+from homeassistant.config_entries import SOURCE_USER
+from homeassistant.const import CONF_HOST
+from homeassistant.core import HomeAssistant
+from homeassistant.data_entry_flow import FlowResultType
+
+from tests.common import MockConfigEntry
+
+
+@pytest.fixture
+def mock_pooldose_client():
+    """Create a mock PooldoseClient."""
+    client = MagicMock()
+    client.device_info = {"SERIAL_NUMBER": "SN123456789"}
+    client.connect = AsyncMock(return_value=RequestStatus.SUCCESS)
+    client.check_apiversion_supported = MagicMock(
+        return_value=(RequestStatus.SUCCESS, {})
+    )
+    client.is_connected = True
+    return client
+
+
+async def test_form_shows_and_creates_entry(
+    hass: HomeAssistant, mock_pooldose_client
+) -> None:
+    """Test that the form is shown and entry is created on valid input."""
+    with patch(
+        "homeassistant.components.pooldose.config_flow.PooldoseClient",
+        return_value=mock_pooldose_client,
+    ):
+        result = await hass.config_entries.flow.async_init(
+            DOMAIN, context={"source": SOURCE_USER}
+        )
+        assert result["type"] == FlowResultType.FORM
+        assert result["step_id"] == "user"
+
+        user_input = {CONF_HOST: "192.168.1.100"}
+        result2 = await hass.config_entries.flow.async_configure(
+            result["flow_id"], user_input
+        )
```

Comment:
```suggestion
        result = await hass.config_entries.flow.async_configure(
            result["flow_id"], {CONF_HOST: "192.168.1.100"}
        )
```
I strongly recommend to re-use `result` as these are easy tests to copy paste, and having them numbered is a PITA